configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -835,7 +830,9 @@ public long createSession(ApplicationId applicationId, TimeoutBudget timeoutBudg
 
     public void deleteExpiredLocalSessions() {
         Map<Tenant, Collection<LocalSession>> sessionsPerTenant = new HashMap<>();
-        tenantRepository.getAllTenants().forEach(tenant -> sessionsPerTenant.put(tenant, tenant.getSessionRepository().getLocalSessions()));
+        tenantRepository.getAllTenants()
+                        .forEach(tenant -> sessionsPerTenant.put(tenant,
+                                                             Collections.unmodifiableCollection(tenant.getSessionRepository().getLocalSessions())));
```

Comment:
I think you want a copy—from what I can see, modifications to the underlying collection will still cause `ConcurrentModificationException` when you iterate through the unmodifiable view, as that just forwards to the underlying collection.

Reply:
Fixed